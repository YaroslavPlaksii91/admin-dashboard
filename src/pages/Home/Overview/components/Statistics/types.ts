export type StatisticsItemType = {
  section: string;
  value: string;
};

export type StatisticsProps = {
  data: StatisticsItemType[];
};
