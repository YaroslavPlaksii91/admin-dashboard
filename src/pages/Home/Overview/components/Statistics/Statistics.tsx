import { FC } from 'react';
import { List, ListItem, Typography, Box } from '@mui/material';

import { StatisticsProps } from './types';

export const Statistics: FC<StatisticsProps> = ({ data }) => {
  return (
    <Box
      sx={{
        width: '342px',
        padding: '8px 0',
        borderLeft: '1px solid',
        borderColor: 'borderColor',
      }}
    >
      <List>
        {data.map(item => (
          <ListItem
            key={item.section}
            sx={{
              display: 'flex',
              flexDirection: 'column',
              padding: '24px 0',
              '&:not(:last-child)': {
                borderBottom: '1px solid',
                borderBottomColor: 'borderColor',
              },
            }}
          >
            <Typography
              component="p"
              sx={{
                fontWeight: 600,
                lineHeight: 1.38,
                letterSpacing: 0.3,
                mb: '6px',
                color: 'secondaryTextColor',
              }}
            >
              {item.section}
            </Typography>
            <Typography
              component="span"
              sx={{
                fontSize: '24px',
                fontWeight: 700,
                letterSpacing: 0.3,
                lineHeight: 1.3,
              }}
            >
              {item.value}
            </Typography>
          </ListItem>
        ))}
      </List>
    </Box>
  );
};
