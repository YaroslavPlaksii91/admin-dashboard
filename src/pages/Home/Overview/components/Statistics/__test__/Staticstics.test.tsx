import { cleanup, render, screen } from '@testing-library/react';

import { Statistics } from '../Statistics';

describe('Statistics component', () => {
  afterEach(cleanup);

  const mockData = [
    { section: 'Section 1', value: '10' },
    { section: 'Section 2', value: '20' },
    { section: 'Section 3', value: '30' },
  ];

  it('should be rendered', () => {
    const { asFragment } = render(<Statistics data={mockData} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders each item in the data', () => {
    render(<Statistics data={mockData} />);

    const listItems = screen.getAllByRole('listitem');

    expect(listItems).toHaveLength(mockData.length);
  });

  it('displays section and value for each item', () => {
    render(<Statistics data={mockData} />);

    mockData.forEach(item => {
      expect(screen.getByText(item.section)).toBeInTheDocument();
      expect(screen.getByText(item.value)).toBeInTheDocument();
    });
  });
});
