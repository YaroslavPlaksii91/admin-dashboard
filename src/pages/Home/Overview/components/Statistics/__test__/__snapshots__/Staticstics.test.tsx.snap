// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics component should be rendered 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1f47q6k"
  >
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-9angok-MuiListItem-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1kbnajf-MuiTypography-root"
        >
          Section 1
        </p>
        <span
          class="MuiTypography-root MuiTypography-body1 css-6228q-MuiTypography-root"
        >
          10
        </span>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-9angok-MuiListItem-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1kbnajf-MuiTypography-root"
        >
          Section 2
        </p>
        <span
          class="MuiTypography-root MuiTypography-body1 css-6228q-MuiTypography-root"
        >
          20
        </span>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-9angok-MuiListItem-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1kbnajf-MuiTypography-root"
        >
          Section 3
        </p>
        <span
          class="MuiTypography-root MuiTypography-body1 css-6228q-MuiTypography-root"
        >
          30
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
